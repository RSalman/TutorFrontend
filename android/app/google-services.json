{
  "project_info": {
    "project_number": "779865862095",
    "firebase_url": "https://prospr-8a50d.firebaseio.com",
    "project_id": "prospr-8a50d",
    "storage_bucket": "prospr-8a50d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:779865862095:android:1b83d4b60becf7dd",
        "android_client_info": {
          "package_name": "com.tutorfrontend"
        }
      },
      "oauth_client": [
        {
          "client_id": "779865862095-4tk4rhhj654t9tepp8c6i5elqlaj6sgg.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBDjtDNEYE5WbvPy3VumSoq1dEDhKFcOH8"
        }
      ],
      "services": {
        "analytics_service": {
          "status": 1
        },
        "appinvite_service": {
          "status": 1,
          "other_platform_oauth_client": []
        },
        "ads_service": {
          "status": 2
        }
      }
    }
  ],
  "configuration_version": "1"
}