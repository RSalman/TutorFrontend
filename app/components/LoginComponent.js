import React, { Component } from 'react';
import { ActivityIndicator, StyleSheet, Text, View, Image, Dimensions, TextInput, TouchableOpacity, Platform } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import FCM, { FCMEvent } from 'react-native-fcm';
import { authenticate, updateEmail, updatePassword } from '../actions/login';
import ErrorView from './ErrorView';
import { handleNotification } from './TempPushNotification';

class LoginComponent extends Component {

  componentDidMount() {
    //Temp(Salman): This should be done once the User has logged in
    //Register event handler to handle push notifications from any screen
    if (Platform.OS === 'android') {
      this.notificationListener = FCM.on(FCMEvent.Notification, async (notif) => {
        handleNotification(notif);
      });
    }
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.successful_authentication)
      Actions.side_menu({ isTutor: nextProps.isTutor, type: 'reset' });
  }

  renderErrorView() {
    if (this.props.error) {
      return (
        <ErrorView error={this.props.error} />
      );
    }
  }

  renderButtonContent() {
    if (this.props.isLoading) {
      return (
        <ActivityIndicator
          animating
          style={styles.spinner}
          size="large"
          color={white}
        />
      );
    }
    return (
      <Text style={styles.buttonText}>Sign In</Text>
    );
  }

  render() {
    return (
      <Image source={background} style={styles.background} resizeMode="cover">
        <View style={styles.container}>
          <View style={styles.markWrap}>
            <Image source={mark} style={styles.mark} resizeMode="contain" />
          </View>
          <View style={styles.wrapper}>
            <View style={styles.inputWrap}>
              <View style={styles.iconWrap}>
                <Image source={personIcon} style={styles.icon} resizeMode="contain" />
              </View>
              <TextInput
                value={this.props.email}
                placeholder="Email"
                placeholderTextColor={grey}
                underlineColorAndroid={transparent}
                style={styles.input}
                onChangeText={(email) => this.props.updateEmail(email)}
              />
            </View>
            <View style={styles.inputWrap}>
              <View style={styles.iconWrap}>
                <Image source={lockIcon} style={styles.icon} resizeMode="contain" />
              </View>
              <TextInput
                value={this.props.password}
                placeholderTextColor={grey}
                placeholder="Password"
                underlineColorAndroid={transparent}
                style={styles.input}
                onChangeText={(password) => this.props.updatePassword(password)}
                secureTextEntry
              />
            </View>
            <TouchableOpacity activeOpacity={0.5}>
              <View>
                <Text style={styles.forgotPasswordText}>Forgot Password?</Text>
              </View>
            </TouchableOpacity>
            <TouchableOpacity activeOpacity={0.5} onPress={() => this.props.authenticate(this.props.email, this.props.password)}>
              <View style={styles.button}>
                { this.renderButtonContent() }
              </View>
            </TouchableOpacity>
          </View>
          { this.renderErrorView() }
          <View style={styles.container}>
            <View style={styles.signupWrap}>
              <Text style={styles.accountText}>Don&apos;t have an account?</Text>
              <TouchableOpacity activeOpacity={0.5} onPress={Actions.signup}>
                <View>
                  <Text style={styles.signupLinkText}>Sign Up</Text>
                </View>
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </Image>
    );
  }
}

const { width, height } = Dimensions.get('window');
const background = require('./img/login1_bg.png');
const mark = require('./img/login1_mark.png');
const lockIcon = require('./img/login1_lock.png');
const personIcon = require('./img/login1_person.png');

const grey = '#D3D3D3';
const lightGrey = '#D8D8D8';
const transparent = 'transparent';
const white = '#FFF';
const darkGrey = '#CCC';
const buttonColor = '#FF3366';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    marginTop: 0,
    padding: 10,
    marginLeft: 15,
    marginRight: 15
  },
  markWrap: {
    flex: 1,
    paddingVertical: 30,
  },
  mark: {
    width: null,
    height: null,
    flex: 1,
  },
  background: {
    width,
    height,
  },
  wrapper: { paddingVertical: 30 },
  inputWrap: {
    flexDirection: 'row',
    marginVertical: 10,
    height: 40,
    borderBottomWidth: 1,
    borderBottomColor: darkGrey
  },
  iconWrap: {
    paddingHorizontal: 7,
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: {
    height: 20,
    width: 20,
  },
  input: {
    flex: 1,
    paddingHorizontal: 10,
    color: white
  },
  button: {
    backgroundColor: buttonColor,
    paddingVertical: 20,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 30,
  },
  buttonText: {
    color: white,
    fontSize: 18,
  },
  forgotPasswordText: {
    color: lightGrey,
    backgroundColor: transparent,
    textAlign: 'right',
    paddingRight: 15,
  },
  signupWrap: {
    backgroundColor: transparent,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  accountText: { color: lightGrey },
  signupLinkText: {
    color: white,
    marginLeft: 5,
  },
  spinner: { height: 18 }
});

const mapStateToProps = (state) => {
  return {
    email: state.login.email,
    password: state.login.password,
    successful_authentication: state.login.successful_authentication,
    error: state.login.error,
    isLoading: state.login.isLoading,
    isTutor: state.login.isTutor,
    user_data: state.login.user_data
  };
};

const mapDispatchToProps = (dispatch) => bindActionCreators({ authenticate, updateEmail, updatePassword }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(LoginComponent);
