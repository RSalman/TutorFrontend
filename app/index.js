import React, { Component } from 'react';
import { StyleSheet,TouchableHighlight, Text, View, Navigator } from 'react-native';

import HomeView from './views/HomeView';
import TutorsComponent from './components/TutorsComponent';
import LeilaFeatureView from './views/LeilaFeatureView';
import SalmanFeatureView from './views/SalmanFeatureView';
import MuraadFeatureView from './views/MuraadFeatureView';

export default class TutoringApp extends Component {
  render() {
    return (
  		<Navigator
        	initialRoute={{screen: 'HomeView'}}
        	renderScene={(route, nav) => {return this.renderScene(route, nav)}}  />
    );
  }

 renderScene(route,nav) {
    switch (route.screen) {
      	case "HomeView":
        	return <HomeView navigator={nav} />
      	case "TutorsComponent":
        	return <TutorsComponent navigator={nav} />
      	case "SalmanFeature":
        	return <SalmanFeatureView navigator={nav} /> 
        case "LeilaFeature":
        	return <LeilaFeatureView navigator={nav} /> 
        case "MuraadFeature":
        	return <MuraadFeatureView navigator={nav} /> 
      }
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});
