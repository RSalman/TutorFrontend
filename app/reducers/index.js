import { combineReducers } from 'redux';
import tutors from './tutors';
import profile from './profile';

const rootReducer = combineReducers({
  tutors: tutors,
  profile: profile
});

export default rootReducer;
