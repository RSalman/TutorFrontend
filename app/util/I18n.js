import I18n from 'react-native-i18n';

import en from './locales/en';
import fr from './locales/fr';

I18n.fallbacks = true;

I18n.translations = {
  en,
  fr
};

global.I18n = I18n;
