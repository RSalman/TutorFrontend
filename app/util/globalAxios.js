import axios from 'axios';
import { Alert, NetInfo } from 'react-native';

axios.defaults.baseURL = 'http://cpserver.eastus.cloudapp.azure.com:2000/api/v1';
axios.defaults.headers.post['Content-Type'] = 'application/json';
axios.defaults.headers.get['Content-Type'] = 'application/json';
axios.defaults.headers.post.Accept = 'application/json';
axios.defaults.headers.get.Accept = 'application/json';

var connection = true;
function connectivityChange(isConnected) {
  connection = isConnected;
}

function showNoInternetDialog(config) {
  return new Promise(function(resolve, reject) {
    Alert.alert(
      I18n.t('globalWifiDialog.noInternetConnection'),
      I18n.t('globalWifiDialog.fixInternet'),
      [
        { text: I18n.t('retry'), onPress: () => resolve(axios(config)) },
      ],
      { cancelable: false }
    );
  });
}

NetInfo.isConnected.addEventListener('change', connectivityChange);

axios.interceptors.response.use(undefined, function(error) {
  return connection ? Promise.reject(error) : showNoInternetDialog(error.config);
});

global.axios = axios;
