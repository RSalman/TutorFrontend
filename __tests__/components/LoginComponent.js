import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import LoginComponent from '../../app/components/LoginComponent';
import '../../app/index';
import '../../app/util/globalAxios';
import '../../app/util/I18n';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

function setup(state) {
  const store = mockStore(state);
  const props = {
    authenticate: jest.fn(),
    store
  };
  return props;
}

describe('<LoginComponent />', () => {
  it('should display login screen', () => {
    const props = setup({
      login: {
        email: 'student@uottawa.ca',
        password: 'complexPassword',
        successful_authentication: false,
        error: '',
        isLoading: false,
        user_data: null
      }
    });
    const tree = renderer.create(
      <LoginComponent {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
