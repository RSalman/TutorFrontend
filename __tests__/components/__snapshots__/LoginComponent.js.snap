exports[`<LoginComponent /> should display login screen 1`] = `
<Image
  resizeMode="cover"
  source={1}
  style={
    Object {
      "height": 1334,
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 0,
        "padding": 10,
      }
    }>
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 30,
        }
      }>
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "width": null,
          }
        } />
    </View>
    <View
      style={
        Object {
          "paddingVertical": 30,
        }
      }>
      <View
        style={
          Object {
            "borderBottomColor": "#CCC",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "marginVertical": 10,
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingHorizontal": 7,
            }
          }>
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            } />
        </View>
        <TextInput
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="#D3D3D3"
          style={
            Object {
              "color": "#FFF",
              "flex": 1,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value="student@uottawa.ca" />
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#CCC",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 40,
            "marginVertical": 10,
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingHorizontal": 7,
            }
          }>
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": 20,
                "width": 20,
              }
            } />
        </View>
        <TextInput
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="#D3D3D3"
          secureTextEntry={true}
          style={
            Object {
              "color": "#FFF",
              "flex": 1,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
          value="complexPassword" />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF3366",
              "justifyContent": "center",
              "marginTop": 30,
              "paddingVertical": 20,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFF",
                "fontSize": 18,
              }
            }>
            Sign In
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 0,
          "padding": 10,
        }
      }>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#D8D8D8",
            }
          }>
          Don\'t have an account?
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#FFF",
                  "marginLeft": 5,
                }
              }>
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Image>
`;
